h2/stream.py

```diff
@@ -1071,30 +1074,26 @@ def _build_hdr_validation_flags(self, events):
         Constructs a set of header validation flags for use when normalizing
         and validating header blocks.
         """
-        try:
-            is_trailer = isinstance(
-                events[0], (_TrailersSent, TrailersReceived)
-            )
-            is_response_header = isinstance(
-                events[0], (_ResponseSent, ResponseReceived)
+        is_trailer = isinstance(
+            events[0], (_TrailersSent, TrailersReceived)
+        )
+        is_response_header = isinstance(
+            events[0],
+            (
+                _ResponseSent,
+                ResponseReceived,
+                InformationalResponseReceived
```

Comment:
I *strongly* recommend we don't do that. The default logic for 1XX responses that user agents can't tolerate is to ignore them, and having them caught in an `isinstance()` check for `ResponseReceived` will cause them in most cases to accidentally be treated as "proper" responses.